src/Core/View.js

```diff
@@ -82,15 +114,30 @@ function _preprocessLayer(view, layer, provider) {
     // probably not the best place to do this
     if (layer.type == 'color') {
         layer.frozen = layer.frozen || false;
-        layer.visible = layer.visible == undefined ? true : layer.visible;
-        layer.opacity = layer.opacity == undefined ? 1.0 : layer.opacity;
-        layer.sequence = 0;
+        _defineLayerProperty(view, layer, 'visible', true);
+        _defineLayerProperty(view, layer, 'opacity', 1.0);
+        layer.addEventListener('layer-changed', () => view.notifyChange(0, true));
+        layer.sequence = layer.sequence == undefined ? 0 : layer.sequence;
     } else if (layer.type == 'elevation') {
         layer.frozen = layer.frozen || false;
+    } else if (layer.type == 'geometry' || layer.type == 'debug') {
+        _defineLayerProperty(view, layer, 'visible', true);
+        layer.addEventListener('layer-changed', (e) => {
+            if (e.visible !== undefined) {
+                if (e.visible) {
+                    view.camera.camera3D.layers.enable(layer.threejsLayer);
+                } else {
+                    view.camera.camera3D.layers.disable(layer.threejsLayer);
+                }
```

Comment:
IMHO these lines:
```js
                if (e.visible) {
+                    view.camera.camera3D.layers.enable(layer.threejsLayer);
+                } else {
+                    view.camera.camera3D.layers.disable(layer.threejsLayer);
+                }
```
should be in the property setter - not in the event listener (the only question is how to get access to the `view`).

And I'd remove the `notifyChange` (for the same reason: it's up to the user)